king_phisher/client/dialogs/campaign_selection.py

```diff
@@ -79,18 +80,70 @@ def __init__(self, *args, **kwargs):
 		self._tv_model.set_sort_column_id(5, Gtk.SortType.DESCENDING)
 		# create and set the filter for expired campaigns
 		self._tv_model_filter = self._tv_model.filter_new()
-		self._tv_model_filter.set_visible_func(self._filter_expired)
+		self._tv_model_filter.set_visible_func(self._filter_campaigns)
 		treeview.set_model(Gtk.TreeModelSort(model=self._tv_model_filter))
+
+		#setup menus for filtering campaigns and load campaigns
+		self.get_popup_filter_menu()
 		self.load_campaigns()
 
-	def _filter_expired(self, model, tree_iter, _):
-		if self.gobjects['checkbutton_show_expired'].get_active():
-			return True
+	def get_popup_filter_menu(self):
+		# create filter menu and menuitems
+		filter_menu = Gtk.Menu()
+		menu_item_expired = Gtk.CheckMenuItem('Expired campaigns')
+		menu_item_user = Gtk.CheckMenuItem('Your campaigns')
+		menu_item_other = Gtk.CheckMenuItem('Other campaigns')
+		self.filter_menu_items = {
+			'expired_campaigns': menu_item_expired,
+			'your_campaigns': menu_item_user,
+			'other_campaigns': menu_item_other
+		}
+		# set up the menuitems and add it to the menubutton
+		for menus in self.filter_menu_items:
+			filter_menu.append(self.filter_menu_items[menus])
+			self.filter_menu_items[menus].connect('toggled', self.signal_checkbutton_toggled)
+			self.filter_menu_items[menus].show()
+
+		try:
+			self.filter_menu_items['expired_campaigns'].set_active(self.config['menu_item_expired'])
+			self.filter_menu_items['your_campaigns'].set_active(self.config['menu_item_user'])
+			self.filter_menu_items['other_campaigns'].set_active(self.config['menu_item_other'])
+		except RuntimeError:
```

Comment:
Under what conditions would this occur? If the config value doesn't exist wouldn't it throw a `KeyError`? Also because they're in the default config they should be added automatically.